python/cudf/cudf/core/column/numerical.py

```diff
@@ -478,7 +478,8 @@ def fillna(
             return super(NumericalColumn, col).fillna(fill_value, method)
 
         if np.isscalar(fill_value):
-            # castsafely to the same dtype as self
+            assert fill_value is not None
```

Comment:
Any other options here? I know this is used for typing purposes, but maybe we should handle `None` more explicitly somehow or something else?

Reply:
Yup - raising a `ValueError` instead both gets rid of the mypy error and makes the code better. FIxed.